src/cartesian/Bar.tsx

```diff
@@ -226,7 +227,7 @@ export class Bar extends PureComponent<Props, State> {
         x,
         y,
         width,
-        height,
+        height: Number.isNaN(height) ? 0 : height,
```

Comment:
This line really is the only relevant change, right? The rest is just refactoring.

Reply:
The height is also used for the Tooltip position a few lines below this one. We should move `const validHeight = Number.isNaN(height) ? 0 : height` up a few lines and then use in both places in the return type.

Reply:
line 194 is actually more relevant. 

line 194:
If we don't have a currentValueScale (aka, current value is null or undefined) then use the baseValueScale (aka whatever y value is computed at first without taking the value into account, and if we don't have that then use undefined and allow `NaN` down the line because something is wrong.

Reply:
will check the tooltip comment out, I'm not sure

Reply:
It turns out that height does not affect the tooltip because when the tooltip isn't shown (since there's no data) these values aren't used. But better safe than sorry so moved the check up.